### PR TITLE
iss12_add adjust_slope_offset function

### PR DESCRIPTION
**Pull request by stephenholleran**
function to adjust a wind speed with a new slope and offset.

I have written lots of tests in a jupyter notebook. These will need to be properly implemented in the test area.

### CODE REVIEW

**Comment by Inder95** [comment]
Hi Stephen, I have added a basic skeleton to add tests for this function and pushed it to the repo. Do check it out. Also I followed this link to set up testing environment on my laptop:
http://blog.thedigitalcatonline.com/blog/2015/05/13/python-oop-tdd-example-part1/

File: brightwind/transform/transform.py [hunk 1]
```diff
@@ -156,7 +156,46 @@ def average_data_by_period(data: pd.Series, period, aggregation_method='mean', f
...
+        df = bw.load_campbell_scientific(bw.datasets.demo_site_data)
+        df['Spd80mS_adj'] = bw.adjust_slope_offset(df.Spd80mS, 0.044, 0.235, 0.04365, 0.236)
+        df[['Spd80mS', 'Spd80mS_adj']]
+
+    """
+    try:
+        return new_slope * ((wspd - current_offset) / current_slope) + new_offset
+    except TypeError as type_error:
+        for arg_value in [current_slope, current_offset, new_slope, new_offset]:
+            if type(arg_value) is not float:
```
Comment (Inder95) [inline brightwind/transform/transform.py]: They can be of type int too.

File: brightwind/transform/transform.py [hunk 2]
```diff
@@ -156,7 +156,46 @@ def average_data_by_period(data: pd.Series, period, aggregation_method='mean', f
...
+                raise TypeError("argument '" + str(arg_value) + "' is not of data type number")
+        if type(wspd) is not int or type(wspd) is not float:
+            if type(wspd) == pd.DataFrame and (wspd.dtypes == object)[0]:
```
Comment (Inder95) [inline brightwind/transform/transform.py]: wspd can be data series too.

File: brightwind/transform/transform.py (hunk 1 above)
Comment (stephenholleran) [reply brightwind/transform/transform.py]: added this

File: brightwind/transform/transform.py (hunk 2 above)
Comment (stephenholleran) [reply brightwind/transform/transform.py]: added Series option too

**Comment by stephenholleran** [comment]
I've added those two items about int and Series.

Once you have accepted this pull request i'll add in the tests as i'd say I am going to get conflicts if I modify your test_transform.py file. Once I get familiar with this process I'll do it.

**Pull request merged**